src/payment/strategies/braintree/braintree-hosted-form.ts

```diff
@@ -133,6 +142,42 @@ export default class BraintreeHostedForm {
         }
     }
 
+    async tokenizeWith3DSRegulationCheck(billingAddress: Address, merchantAccountId: string): Promise<Pick<BraintreeHostedFieldsTokenizePayload, 'authenticationInsight'> & NonceInstrument> {
+        if (!this._cardFields) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        try {
+            const { authenticationInsight, nonce } = await this._cardFields.tokenize(omitBy({
+                authenticationInsight: {
+                    merchantAccountId,
+                },
+                billingAddress: billingAddress && this._mapBillingAddress(billingAddress),
+                cardholderName: this._cardNameField?.getValue(),
+            }, isNil));
+
+            this._formOptions?.onValidate?.({
+                isValid: true,
+                errors: {},
+            });
+
+            return { authenticationInsight, nonce };
+        } catch (error) {
+            const errors = this._mapTokenizeError(error);
+
+            if (errors) {
+                this._formOptions?.onValidate?.({
+                    isValid: false,
+                    errors,
+                });
+
+                throw new PaymentInvalidFormError(errors as PaymentInvalidFormErrorDetails);
```

Comment:
Type guard please

Reply:
We have the same row in two methods above. Is it ok to make it in another pr, like braintree hosted form class refactoring?